cleanup/DeleteBasedonProperties/artifactdeletebasedonproperty.groovy

```diff
@@ -0,0 +1,157 @@
+/**
+    Prunes data from specified repos using method defined in repo/folder properties
+*/
+// refs
+import org.artifactory.build.promotion.PromotionConfig
+import org.artifactory.build.staging.ModuleVersion
+import org.artifactory.build.staging.VcsConfig
+import org.artifactory.exception.CancelException
+import org.artifactory.repo.RepoPathFactory
+import org.artifactory.request.Request
+import org.artifactory.util.StringInputStream
+import org.artifactory.resource.ResourceStreamHandle
+import groovy.json.JsonSlurper
+import groovy.transform.Field
+import org.artifactory.fs.ItemInfo
+import org.artifactory.repo.RepoPath
+import static com.google.common.collect.Multimaps.forMap
+// jobs to trigger based on cron or interval
+jobs {
+    // Every Min Run cron Regex  depreciation run
+    dailyDataDepreciation(cron: "0 0/1 * 1/1 * ? *") {
+        depreciateData("repositoryname-local")
+    }
+}
+def depreciateData(repoName){
+    
+    log.info "Starting data depreciation job for repository " + repoName // This will log in the LogFile
+    
+    // search given repo name for folders with property DataDeprecation = true 
+    def repoRootPath = RepoPathFactory.create(repoName, "/")
+    List<RepoPath> deprecatePaths = searches.itemsByProperties(forMap(['DataDeprecation': 'true']))
+    // call the corresponding method to handle deprecation
+    for(RepoPath repoPath : deprecatePaths){
+        log.info "Depreciating data for path " + repoPath
+        
+
+        String deprecateMethodStr = repositories.getProperty(repoPath, 'DataDeprecationMethod')   //version:4 This is the Property on the artifact
+if (deprecateMethodStr !=null)
+{
+    String[] depreciateMethodOpts = deprecateMethodStr.split(':')   //  (version) : (4) 
+
+    String[] depreciateMethodArgs = depreciateMethodOpts[1]
+moduleMajorVersion(repoPath,repoName,depreciateMethodArgs)
+  
+}else{
+    println "No Such Property Defined on the folder"
+}
+      
+    }
+}
+/**
+    Deprecate files by Module name and major version, removing by build number
+    Keep last x number of semantic versions of an Artifact
+    args is assumed to be a single integer value (but as a string) for number of files to keep
+*/
+def moduleMajorVersion(RepoPath repoPath, repoName,String[] args){
```

Comment:
`args` is always hosting a single value. There is no need for it to be a list. Also, the parameter name should indicate the function of the parameter.